packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx

```diff
@@ -235,28 +254,24 @@ export const ReleaseDetailsLayout = ({
                       direction="column"
                       padding={1}
                     >
-                      <CheckPermissions permissions={PERMISSIONS.update}>
-                        <PopoverButton onClick={openReleaseModal}>
-                          <PencilIcon />
-                          <Typography ellipsis>
-                            {formatMessage({
-                              id: 'content-releases.header.actions.edit',
-                              defaultMessage: 'Edit',
-                            })}
-                          </Typography>
-                        </PopoverButton>
-                      </CheckPermissions>
-                      <CheckPermissions permissions={PERMISSIONS.delete}>
-                        <PopoverButton onClick={openWarningConfirmDialog}>
-                          <TrashIcon />
-                          <Typography ellipsis textColor="danger600">
-                            {formatMessage({
-                              id: 'content-releases.header.actions.delete',
-                              defaultMessage: 'Delete',
-                            })}
-                          </Typography>
-                        </PopoverButton>
-                      </CheckPermissions>
+                      <PopoverButton disabled={!canUpdate} onClick={openReleaseModal}>
+                        <PencilIcon />
+                        <Typography ellipsis>
+                          {formatMessage({
+                            id: 'content-releases.header.actions.edit',
+                            defaultMessage: 'Edit',
+                          })}
+                        </Typography>
+                      </PopoverButton>
+                      <PopoverButton disabled={!canDelete} onClick={openWarningConfirmDialog}>
+                        <TrashIcon />
+                        <Typography ellipsis textColor="danger600">
+                          {formatMessage({
+                            id: 'content-releases.header.actions.delete',
+                            defaultMessage: 'Delete',
+                          })}
```

Comment:
There's something wrong with the delete icon, it's grey when it should be red if the action is enabled.

I think you may need to do `<Icon as{TrashIcon} color="danger600" />` here

Then in StyledFlex I would only ensure that you override the color to be gray when it's disabled, but to not set any styles when it's not disabled